src/main/scala/com/metamx/rainer/CommitStorage.scala

```diff
@@ -83,6 +83,32 @@ object CommitStorage extends Logging
     }
   }
 
+  def withPreSaveHook[ValueType](delegate: CommitStorage[ValueType])(hook: Commit[ValueType] => Unit) = {
+    new CommitStorage[ValueType]
+    {
+      override def start() {
+        delegate.start()
+      }
+
+      override def stop() {
+        delegate.stop()
+      }
+
+      override def save(commit: Commit[ValueType]) {
+        hook(commit)
+        delegate.save(commit)
```

Comment:
Propagate exception to the top and handle it by code which called `save` method or in `hook` itself.